pennylane_lightning/lightning_qubit/_adjoint_jacobian.py

```diff
@@ -14,6 +14,9 @@
 r"""
 Internal methods for adjoint Jacobian differentiation method.
 """
+from __future__ import annotations
```

Comment:
How is this in use?

Reply:
[Main post](https://github.com/PennyLaneAI/pennylane-lightning/pull/923#discussion_r1777524389)
> Added future to improve the typing for `LightningGPUStateVector` without loading the module again.

I followed this [idea](https://stackoverflow.com/questions/61544854/from-future-import-annotations). The objective was to prevent the import multiple times of the classes `state_vector`, `measurements`, and `Adj_Jac` in all the device files.